### PR TITLE
XGBoost model added for Chicago Food Inspection

### PR DESCRIPTION
**Pull request by socratesk**
The XGBoost model performs better than RandomForest model in both Train and Test data sets. The comparison metrics can be found at the bottom of the "31_xgboost_model_evaluation.R"

#Random Forest: Time difference of **6.554264** days
#XGBoost: Time difference of **7.790698** days

### CODE REVIEW

**Comment by tomschenkjr** [comment]
@socratesk - thank you! This looks very promising. The XGBoost obviously is a good candidate since the 7.8 day improvement is greater than the 7.4 day improvement we see with our logit model.

We are working on a couple of other projects and will be doing a code review on your contribution as soon as we can. We will be validating your results as a second pair of eyes and will follow-up with any questions. If the results hold, we will incorporate your contributions to the model that drives food inspections in the city.

Again, thank you and will be in touch soon!

**Comment by socratesk** [comment]
Hi Tom,

I prepared an App and uploaded it in ShinyApps for **Chicago Food Inspection**. You may goto the [app using this link](https://socrates.shinyapps.io/chicagofoodinspection/). Since it is uploaded under free-tier some of the models may get disconnected however most of the functionalities and models will still work. Will migrate it under paid-tier shortly.

Let me know what you think.

Thanks!

File: CODE/30_xgboost_model.R [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+#table(test.target)
+
+set.seed(1)
+
+# 5% of test data is taken for validation purpose (733+120 = 853 [5% of 17075 - Train set] )
+h <- c( sample(which(train.target == 0), 733), sample(which(train.target == 1), 120))
+
+# Create validation data set based on the above rows
+dval   <- xgb.DMatrix(data = data.matrix(train[h, ]), label = train.target[h])
+dtrain <- xgb.DMatrix(data = data.matrix(train), label = train.target)
```
[the remark this answers is not in the document]
Comment (socratesk) [reply CODE/30_xgboost_model.R]: `dtrain <- xgb.DMatrix(data = data.matrix(train[-h, ]), label = train.target[-h])`
This is correct

File: CODE/30_xgboost_model.R [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+# RMSE: 0.3596287; 1200 rounds; eta = 0.03;  subsample = 0.70; colsample_bytree = 0.70 - multi:softprob
+# RMSE: 0.3578677; 1200 rounds; eta = 0.02;  subsample = 0.70; colsample_bytree = 0.70 - multi:softprob
+# RMSE: 0.3576444; 1100 rounds; eta = 0.02;  subsample = 0.70; colsample_bytree = 0.70 - multi:softprob
+# RMSE: 0.3559322; 1100 rounds; eta = 0.02;  subsample = 0.70; colsample_bytree = 0.70 - multi:softprob
+# RMSE: 0.3563522;  500 rounds; eta = 0.02;  subsample = 0.75; colsample_bytree = 0.75 - multi:softprob
+# RMSE: 0.3563522; 1000 rounds; eta = 0.02;  subsample = 0.75; colsample_bytree = 0.75 - multi:softprob
+# RMSE: 0.3792566; 1200 rounds; eta = 0.002; subsample = 0.75; colsample_bytree = 0.75 - multi:softprob
+
+## ATTACH PREDICTIONS TO DAT
+y_pred_mm <- predict(xgbmodel, data.matrix(mm), ntree = xgbmodel$bestInd)
```
[the remark this answers is not in the document]
Comment (socratesk) [reply CODE/30_xgboost_model.R]: It seems persisting the model in hard disk and loading the same looses the `bestind` attribute.
[https://github.com/dmlc/xgboost/issues/1415](https://github.com/dmlc/xgboost/issues/1415)

By the way, I did not tune all the parameters to get the maximum performance. Primarily, I was focusing onto `eta, subsample`, and `colsample_bytree`
